client/src/dataset/components/ScanSize.tsx

```diff
@@ -0,0 +1,70 @@
+import { Field } from "formik";
+import * as React from "react";
+import { Button, Form, FormFieldProps } from "semantic-ui-react";
+
+interface ScanSizeProps {
+    value: string,
+    minScan: number,
+    maxScan: number,
+    setFieldValue: (field: string, value: any, shouldValidate?: boolean) => void,
+}
+
+const ScanSize: React.FC<ScanSizeProps> = ({ value, minScan, maxScan, setFieldValue }) => {
+
+    const [scanSize, setScanSize] = React.useState(value.split(",").length>=minScan? value.split(","): new Array(minScan).fill(""));
+
+    const scanRefsArray = React.useRef<FormFieldProps[]>([]);
+
+    const onScanSizeChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+      const idx = parseInt(e.target.name.split("_")[2], 10);
+      scanSize[idx] = e.target.value.toString();
+      setScanSize(scanSize);
```

Comment:
Hmm, I'm not sure this is right. You are modifying an element of `scanSize`, and then call `setScanSize` with the "same" `scanSize` object as it was before. "same" meaning same object identity here. The `useState` docs say this: "If your update function returns the exact same value as the current state, the subsequent rerender will be skipped completely." As I understand, this is a shallow equality, so it will only check the identity of the object passed, not the values contained inside, like array elements. Then, calling `setScanSize` does not trigger a re-render, which may also be the reason you had to fiddle with the `key` in the forms.

If you instead return a copy of the array, with the new values, you should get a re-render when calling the setter. [Here is a codesandbox that demonstrates the issue:](https://codesandbox.io/s/vigilant-turing-bs9ej?file=/src/App.js) clicking the "set a" button does not immediately cause a re-render, clicking "set value", which creates a new object, triggers a re-render.

So generally, the value returned from `useState` should be considered immutable, even if JavaScript allows you to mutate it.

Reply:
> `setScanSize` does not trigger a re-render, which may also be the reason you had to fiddle with the key in the forms

Yes, that is actually the reason. I'll fix this, thanks!